internal/connector/connector.go

```diff
@@ -151,6 +162,25 @@ func Run(ctx context.Context, options Options) error {
 			logging.Errorf("Unauthorized error; token invalid or expired. exiting.")
 			cancel()
 		},
+		ObserveFunc: func(start time.Time, request *http.Request, response *http.Response, err error) {
+			errorLabel := ""
+			if err != nil {
+				errorLabel = err.Error()
```

Comment:
Won't this be too high cardinality, since these are effectively infinite error strings?

I was thinking of just a `true` or `false` value for the error label.

Reply:
Technically yes. My concern with setting just `true` or `false` is it provides no information as to why it failed. With the status code, there's at least some context. Setting a `true` or `false` error is almost like setting status to `-1` on error instead of a meaningful HTTP code

Reply:
That's true. We could attempt to detect the error and reduce it to a few well known constant values, but I'm not sure it's worth the effort right now. We can always add that later as a separate `errorClass` or some other label.

Generally I would not expect metrics to tell me about the error. If I see the number of errors increasing I would go to logs to figure out more details about the errors.

Reply:
That's true. It's nice when metrics tell you what the problem is but not necessary. In this case, I'd prefer setting `status` to `-1` since `status` and `error=true` are (probably) mutually exclusive; adding a separate labels seems unnecessary